### PR TITLE
[New Model] Week-Based Material Demand (Demand and Capacity Management)

### PR DESCRIPTION
**Pull request by LukasHeimann**
## Description
This aspect contains data about a Material Demand from a given customer towards a particular supplier. It is necessary for the use case of Demand and Capacity Management (DCM), representing the "Demand" part of the use case. A parallel request has been opened for the corresponding model of a Capacity Group as shared by the supplier (https://github.com/eclipse-tractusx/sldt-semantic-models/pull/91). Both models co-evolve.

In DCM, customers along a supply chain want to make the demands for their materials at certain locations with certain demand categories available for their suppliers. Demand data is provided on a weekly basis for the next two years. Suppliers, in return make their capacities for certain demand series (based on material, location and demand category) transparent to the respective customers.

Compare
- https://github.com/catenax/BAMMmodels/issues/182
- https://github.com/catenax/BAMMmodels/pull/207

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the BAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar bamm-cli.jar -i \<path-to-aspect-model\> -v ). The  BAMM CLI is available [here](https://openmanufacturingplatform.github.io/sds-documentation/sds-developer-guide/dev-snapshot/tooling-guide/bamm-cli.html) and in [GitHub](https://github.com/OpenManufacturingPlatform/sds-sdk/releases)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "name" and "description"** in English language. 
- [x] **no duplicate names or preferredNames** within an Aspect (e.g. a Property and the referenced Characteristic should not have the same name)
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the BAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/v1.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'

### CODE REVIEW

**Review by bs-jokri — APPROVED** [review]
LGTM

**Comment by LukasHeimann** [comment]
Thank you for reviewing and approving! Please wait with merging until the use case gave their final okay as well.

**Comment by bs-jokri** [comment]
@LukasHeimann this is eaxctly the process. Now you check back with the use case if the model is ok and good for release (MS3) if this is the case, you put the model on the agenda of the CTSM meeting and it is officially MS3 approved and finally released.

**Comment by petyo-iml** [comment]
@LukasHeimann happy to see the model has been submitted for review. Nevertheless, I still see an issue with the aspect's name.

The `MaterialDemand` aspect is modeled following the requirements of the DCM use case. Still there are other UCs currently discussing the exchange of material demands in one way or another. IMHO, publishing an aspect model named `MaterialDemand` under `io.catenax` implies, that it describes any material demand within Catena-X in a generic / abstract way. That is not the case.

I can imagine two possible solutions:
- move the aspect model to a use case specific package / namespace (e.g. `io.catenax.dcm`)
OR
- rename the aspect model (e.g. `DcmMaterialDemand`)

**Comment by LukasHeimann** [comment]
Hi @petyog, there's still a small thing missing, I think demand categories have been renamed, but the main thing should be good to go.

I totally agree with your comment. I remember half a year ago wanting to use a `dcm` namespace as well, but that was rejected by the semantic modelling team: https://github.com/catenax/BAMMmodels/pull/208#discussion_r952946368. Perhaps the overall opinion has changed now?

File: io.catenax.week_based_material_demand/1.0.0/WeekBasedMaterialDemand.ttl [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    bamm:preferredName "Demand Series Set"@en;
+    bamm:description "A set of demand series belonging to a certain material demand. Each demand series in the set must have a unique combination of customer location and demand category."@en;
+    bamm:dataType :DemandSeries.
+:BPNL a bamm:Characteristic;
+    bamm:preferredName "Business Partner Number Legal Entity"@en;
+    bamm:description "BPNL represents the legal entity of an organization participating in Catena-X."@en;
+    bamm:dataType xsd:string.
+:expectedSupplierLocation a bamm:Property;
+    bamm:preferredName "Expected Supplier Location"@en;
+    bamm:description "The BPNS number of the site from where the customer expects the supplier to fulfill the demands of the demand series. The BPNS number of the site from where the customer expects the supplier to fulfill the demands of the demand series. The value is used for informational purposes only and is therefore not binding for the supplier."@en;
```
Comment (petyog) [inline io.catenax.week_based_material_demand/1.0.0/WeekBasedMaterialDemand.ttl:93]: @LukasHeimann the description of the `expectedSupplierLocation` property is redundant. Can you fix that as well before the model is submitted for final approval? Thx!

Comment (LukasHeimann) [reply io.catenax.week_based_material_demand/1.0.0/WeekBasedMaterialDemand.ttl]: Good catch, was a mistake during copy-paste from the email

Comment (petyog) [reply io.catenax.week_based_material_demand/1.0.0/WeekBasedMaterialDemand.ttl]: Always happy to review every single bit and byte 😇

**Comment by petyog** [comment]
@bs-jokri are there still any open tasks / pending checks or can this PR be merged? We need the final link to the `WeekBasedMaterialDemad.ttl` file for the corresponding standardization request (CX-0047).

**Comment by bs-jokri** [comment]
@petyog  same here. automated tests have finished. Model is merged to main branch.

**Pull request merged**